src/nunit.analyzers/UseCollectionConstraint/UseCollectionConstraintCodeFix.cs

```diff
@@ -37,18 +37,19 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
 
             var node = root.FindNode(context.Span);
 
-            if (!(node is ArgumentSyntax actualArgument) ||
-                !(actualArgument.Expression is MemberAccessExpressionSyntax actualExpression) ||
-                !(actualArgument.Parent is ArgumentListSyntax argumentList) ||
-                !(argumentList.Arguments.Count > 1 && argumentList.Arguments[1] is ArgumentSyntax constraintArgument))
+            if (node is not ArgumentSyntax actualArgument ||
+                actualArgument.Expression is not MemberAccessExpressionSyntax actualExpression ||
+                actualArgument.Parent is not ArgumentListSyntax argumentList ||
+                argumentList.Arguments.Count == 1 ||
+                argumentList.Arguments[1] is not ArgumentSyntax constraintArgument)
```

Comment:
Shouldn't the first line be `argumentList.Arguments.Count <= 1 ||` ?

Before this change we would return if `argumentList.Arguments.Count == 0` now I think we would fail as we are accessing index 1 (in the second line)

Reply:
If it helps with readability, yes, I'm in favour.

Practically it isn't necessary.
The Diagnostic is not even raised if there are less than two arguments (see `AssertHelper.TryGetActualAndConstraintOperations`)
The line above ensures that the node we are dealing with is an `ArgumentSyntax` with the `argumentList` being its `Parent`. This implies that there is at least one argument.

I will update it.
I also have to fix the rebase issues.

Reply:
Thanks 👍